### PR TITLE
Update event-sources to not use fallback logger

### LINKED ISSUE #332: Kafka adapter uses the fallback logger
Looked into this a bit now.  Yes, that's what the logs look like.  It appears that both prior logging messages and current ones are using the fallback logger (which is why we have the extra output).

Testing this quickly locally by commenting out the 
`logger := logging.FromContext(context.TODO())`
line and adding in
```
logger := zap.NewExample()
defer logger.Sync()
```
Seems to produce the logging format you describe.

May I suggest that, due to this being a preexisting issue, we leave it for another, more focused Issue/PR regarding the use of the fallback logger?

_Originally posted by @lberk in https://github.com/knative/eventing-sources/pull/329/files#r271099097_

### PR DESCRIPTION
**Pull request by lberk**
Fixes #332

Currently, event-sources don't properly set, or receive their loggers
from context.  Change this to set the initial logger in both the
event-source, as well as the underlying receiver.

### CODE REVIEW

File: pkg/controller/sdk/reconciler.go [hunk 1]
```diff
@@ -46,8 +47,8 @@ var _ reconcile.Reconciler = &Reconciler{}
 // Reconcile compares the actual state with the desired, and attempts to
 // converge the two.
 func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	ctx := context.TODO()
-	logger := logging.FromContext(ctx)
+	ctx := logging.WithLogger(context.TODO(), r.logger.With(zap.Any("request", request)))
+	logger := r.logger
```
Comment (Harwayne) [inline pkg/controller/sdk/reconciler.go]: Shouldn't this be `logger := logging.FromContext(ctx)`? As-is, you are missing the `request` parameter added on the previous line.

File: cmd/manager/main.go [hunk 2]
```diff
@@ -18,17 +18,33 @@ package main
...
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
 )
 
+const (
+	ConfigMapNameEnv = "CONFIG_LOGGING_NAME"
```
Comment (Harwayne) [inline cmd/manager/main.go]: Is this env var set anywhere?

File: pkg/controller/sdk/reconciler.go (hunk 1 above)
Comment (lberk) [reply pkg/controller/sdk/reconciler.go]: Fixed in commit b18e66e

File: cmd/manager/main.go (hunk 2 above)
Comment (lberk) [reply cmd/manager/main.go]: Fixed in commit b18e66e and dropped the various CONFIG_LOGGING_NAME references, just set the string directly based on the source.

**Pull request merged**